File: app/src/main/java/me/gabeg/sicksends/trad/SsTradViewModel.java
package me.gabeg.sicksends.trad;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemViewModel;

/**
 * Trad view model.
 */
public class SsTradViewModel
	extends SsProblemViewModel<SsTradRepository, SsTrad>
{

	/**
	 */
	public SsTradViewModel(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemViewModel#buildRepository(Application)
	 */
	public void buildRepository(Application app)
	{
		this.mRepository = new SsTradRepository(app);
	}

}

File: app/src/main/java/me/gabeg/sicksends/trad/SsTradViewModel.kt
package me.gabeg.sicksends.trad

import androidx.lifecycle.SavedStateHandle
import dagger.hilt.android.lifecycle.HiltViewModel
import me.gabeg.sicksends.problem.SsGenericProblemViewModel
import javax.inject.Inject

/**
 * Trad view model.
 */
@HiltViewModel
class SsTradViewModel @Inject constructor(
	savedStateHandle: SavedStateHandle,
	repo : SsTradRepository)
		: SsGenericProblemViewModel<SsTradProblem>(savedStateHandle, repo)
